odl/util/utility.py

```diff
@@ -163,6 +185,145 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def default_dtype(impl, field):
```

Comment:
I have implemented the static methods. With those, the code would look like this:
```python
dspc_type = dspace_type(fspace, impl)
dtype = dspc_type.default_dtype(fspace.field)
```
which is more intelligent and extensible. We need that additional step to infer the class first and then get the default type. Maybe that's okay.